controllers/configmap_controller.go

```diff
@@ -170,6 +176,61 @@ func (r *ConfigMapReconciler) reconcileServices(ctx context.Context, windowsServ
 	return nil
 }
 
+// removeOutdatedServicesConfigMaps deletes any outdated services ConfigMaps, if all nodes have moved past that version
+func (r *ConfigMapReconciler) removeOutdatedServicesConfigMaps(ctx context.Context) error {
+	nodes := &core.NodeList{}
+	if err := r.client.List(ctx, nodes, client.MatchingLabels{core.LabelOSStable: "windows"}); err != nil {
+		return err
+	}
+	versionAnnotations := getVersionAnnotations(nodes.Items)
+
+	servicesConfigMaps, err := r.listServiceConfigMaps(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "unable to retrieve list of services ConfigMaps")
+	}
+	for _, cm := range servicesConfigMaps {
+		cmVersion := strings.TrimPrefix(cm.Name, servicescm.NamePrefix)
+		if cmVersion == version.Get() {
+			// Do not delete the services ConfigMap tied to the current WMCO version
+			continue
+		}
+		// Remove any services ConfigMap tied to a WMCO version that no Windows nodes are at anymore
+		if !isTiedToRelevantVersion(cmVersion, versionAnnotations) {
+			if err := r.client.Delete(ctx, &cm); err != nil {
+				return errors.Wrapf(err, "could not delete outdated services ConfigMap %s", cm.Name)
+			}
+			r.log.Info("Deleted outdated resource", "ConfigMap",
+				kubeTypes.NamespacedName{Namespace: r.watchNamespace, Name: cm.Name})
+		}
+	}
+	return nil
+}
+
+// isTiedToRelevantVersion checks if the given version is in the given map of WMCO versions
+func isTiedToRelevantVersion(v string, versions map[string]struct{}) bool {
+	for version := range versions {
+		if v == version {
+			return true
+		}
+	}
+	return false
+}
+
+// listServiceConfigMaps returns a list of all windows-services ConfigMaps in the operator namespace
+func (r *ConfigMapReconciler) listServiceConfigMaps(ctx context.Context) ([]core.ConfigMap, error) {
+	watchNamespaceCMs := &core.ConfigMapList{}
+	if err := r.client.List(ctx, watchNamespaceCMs, &client.ListOptions{Namespace: r.watchNamespace}); err != nil {
```

Comment:
Can you use [FieldSelector](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/) in `client.ListOptions` prepare a query that fetches all `windows-services` ConfigMap?

Reply:
Ideally, we should filter on the server-side. What about introducing an [annotation](https://github.com/openshift/windows-machine-config-operator/blob/61e24e9f8fcb31a451ea1b2408c7382821f4a78f/pkg/windows/windows.go#L92)?

Reply:
Correct, I'm fine with looping the result set.

Reply:
I'm ok with how it is